### PR TITLE
[tree view] Add `slots` and `slotProps` on the Tree View components

### PR DESCRIPTION
**Pull request by flaviendelangle**
- [x] Replace `TreeViewProps["defaultExpandIcon"]` with `TreeViewProps["slots"]["expandIcon"]`
- [x] Replace `TreeViewProps["defaultCollapseIcon"]` with `TreeViewProps["slots"]["collapseIcon"]`
- [x] Replace `TreeItemProps["expandIcon"]` with `TreeItemProps["slots"]["expandIcon"]`
- [x] Replace `TreeItemProps["collapseIcon"]` with `TreeItemProps["slots"]["collapseIcon"]`
- [x] Add a default value to `TreeItemProps["slots"]["expandIcon"]`
- [x] Add a default value to `TreeItemProps["slots"]["collapseIcon"]`

I still need to migrate `TransitionComponent`, `ContentComponent`, `endIcon` and `icon` but I did not wanted this PR to be to hard to review

### CODE REVIEW

**Review by LukasTy — APPROVED** [review]
This is a really nice improvement and a step towards having as similar API between products as possible. 👏 💯

File: packages/x-tree-view/src/TreeItem/TreeItem.tsx [hunk 1]
```diff
@@ -12,6 +14,16 @@ import { TreeItemOwnerState, TreeItemProps } from './TreeItem.types';
 import { useTreeViewContext } from '../internals/TreeViewProvider/useTreeViewContext';
 import { DefaultTreeViewPlugins } from '../internals/plugins';
 
+const TreeViewExpandIcon = createSvgIcon(
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply packages/x-tree-view/src/TreeItem/TreeItem.tsx:17]: Sure :+1:

File: docs/data/tree-view/simple-tree-view/customization/CustomAnimation.js [hunk 2]
```diff
@@ -32,11 +29,6 @@ export default function CustomAnimation() {
     <SimpleTreeView
       aria-label="customized"
       defaultExpandedNodes={['1']}
-      defaultCollapseIcon={
```
Comment (flaviendelangle) [inline docs/data/tree-view/simple-tree-view/customization/CustomAnimation.js:35]: @noraleonte I removed the custom icon here to focus solely on the transition feature
Is that OK for you?

File: docs/data/tree-view/simple-tree-view/customization/customization.md [hunk 3]
```diff
@@ -15,7 +15,7 @@ waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
 
 ### Custom icons
 
-Use the `defaultCollapseIcon`, `defaultExpandIcon`, and `defaultEndIcon` props to customize the Tree View icons.
+Use the `collapseIcon` slot, the `expandIcon` slot and the `defaultEndIcon` prop to customize the Tree View icons.
```
Comment (flaviendelangle) [inline docs/data/tree-view/simple-tree-view/customization/customization.md:18]: This sentence will be better once `defaultEndIcon` becomes a slot as well

File: docs/data/tree-view/simple-tree-view/customization/CustomAnimation.js (hunk 2 above)
Comment (noraleonte) [reply docs/data/tree-view/simple-tree-view/customization/CustomAnimation.js:35]: Sure!

**Pull request merged**